pkg/volume/csi/csi_mounter.go

```diff
@@ -218,10 +202,11 @@ func (c *csiMountMgr) SetUpAt(dir string, mounterArgs volume.MounterArgs) error
 	}
 
 	// create target_dir before call to NodePublish
-	if err := os.MkdirAll(dir, 0750); err != nil && !corruptedDir {
-		return errors.New(log("mounter.SetUpAt failed to create dir %#v:  %v", dir, err))
+	parentDir := filepath.Dir(dir)
```

Comment:
Can we add a unit test to make sure we're creating the right directories?

Reply:
Added